pip/pip-382.md

```diff
@@ -0,0 +1,48 @@
+# PIP-382: Add a label named reason for topic_load_failed_total
+
+# Background knowledge
+
+Pulsar has a metric that indicates load topic failed: `topic_load_failed_total`, it will be increased at the following cases
+- The target bundle in unloading.
+- Failed to load policies.
+- Failed to load up Managed Ledger.
+- Failed to read Metadata store.
+- Topic initialize fails, such as failed to re-build deduplication info.
+- Topic load timeout.
+- Others.
+
+# Motivation & Goals
+
+Adding an additional label of the metric `topic_load_failed_total` may let us know what error happened fastly, so we can fix the issue fastly.
+
+### Metrics
+
+Add a label named reason for topic_load_failed_total
+- label name: `reason`
+- label values:
+  - `bundle_unloading`
+  - `failed_load_policies`
+  - `failed_load_ml`
+  - `failed_access_metadata_store`
+  - `failed_init`
```

Comment:
This refers to the failure in what scenario?

Reply:
I know, here:

https://github.com/apache/pulsar/blob/fc0e4e3fe0fa14d9ac1361871edc53957625fe29/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java#L1774-L1775